utils/src/main/kotlin/Utils.kt

```diff
@@ -567,3 +571,46 @@ fun Throwable.showStackTrace() {
     // https://discuss.kotlinlang.org/t/if-operator-in-function-expression/7227.
     if (printStackTrace) printStackTrace()
 }
+
+/**
+ * Return a Closable which temporarily moves the given directories to a temporary directory on initialization. On close
+ * it moves them back to their original location overwriting any conflicting existing directory.
+ */
+fun stashDirectories(vararg directories: File): Closeable = StashDirectory(setOf(*directories))
+
+private class StashDirectory(directories: Set<File>) : Closeable {
+    private val stashedDirectories = mutableMapOf<File, File>()
+
+    init {
+        directories.forEach {
+            stashDir(it)
+        }
+    }
+
+    override fun close() {
+        HashMap(stashedDirectories).forEach {
```

Comment:
I'd suggest to instead implement this as

        val i = stashedDirectories.iterator()
        while (i.hasNext()) {
            val (originalDir, tempDir) = i.next()

            originalDir.safeDeleteRecursively()

            log.info { "Moving back directory from '${tempDir.absolutePath}' to '${originalDir.absolutePath}'." }

            Files.move(tempDir.toPath(), originalDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
            i.remove()
        }

That solves several things at one: No need to warp in `HashMap`, and due to inlining key / value lookup is done at the same time as iterating.